src/main/java/com/team/piickle/ballot/service/BallotService.java

```diff
@@ -1,8 +1,163 @@
+                    .collect(Collectors.toList());
+        }
+        return randomBallotTopicsExcludeCompleted;
+    }
+
+    private List<BallotTopic> getTopicListForGuest(String userId) {
+        if (userId == null) {
+            return ballotTopicRepository.findAllOrderByOrder(PageRequest.of(0, 4));
+        }
+        return new ArrayList<>();
+    }
+
+    public BallotStatusDto getBallotStatusAndUserSelect(String ballotTopicId, String userEmail) {
+        BallotTopic ballotTopic = ballotTopicRepository.findById(ballotTopicId)
+                .orElseThrow(() -> new GeneralException("올바르지 않은 투표 주제 id 입니다."));
+        List<BallotItem> ballotItems = ballotItemRepository.findByBallotTopicId(new ObjectId(ballotTopicId));
+        String userId = "";
+        if (!userRepository.findByEmail(userEmail).isEmpty()) {
+            userId = userRepository.findByEmail(userEmail).get().getId();
+        }
+        String userSelectId = "";
+        if (!ballotResultRepository.findByBallotTopicIdAndUserId(new ObjectId(ballotTopicId), new ObjectId(userId)).isEmpty()) {
+            userSelectId = ballotResultRepository.findByBallotTopicIdAndUserId(new ObjectId(ballotTopicId), new ObjectId(userId))
+                    .get()
+                    .getId();
+        }
+        int ballotCount = ballotResultRepository.findByBallotTopicId(ballotTopicId).size();
+        String finalUserSelectId = userSelectId;
+        List<BallotStatusDto.BallotItem> ballotItemWithStatusList = ballotItems.stream()
+                .map(value -> {
+                    int status = 0;
+                    if (!finalUserSelectId.equals("")) {
+                        status = Calculation.getRatio(ballotCount, ballotResultRepository.findByBallotItemId(value.getId()).size());
+                    }
+                    return BallotStatusDto.BallotItem.builder()
+                            .id(value.getId())
+                            .status(status)
+                            .content(value.getName())
+                            .build();
+                }).collect(Collectors.toList());
+        return BallotStatusDto.builder()
+                .ballotItems(ballotItemWithStatusList)
+                .ballotTopic(BallotStatusDto.BallotTopic
+                        .builder()
+                        .id(ballotTopicId)
+                        .ballotTopicContent(ballotTopic.getTopic())
+                        .build())
+                .userSelect(finalUserSelectId)
+                .beforeTopicId(getLargestOrderTopicIdLessThan(ballotTopic.getOrder()))
+                .nextTopicId(getSmallestOrderTopicIdGreaterThan(ballotTopic.getOrder()))
+                .build();
+    }
+
+    private String getSmallestOrderTopicIdGreaterThan(long order) {
+        List<BallotTopic> findBallotTopic = ballotTopicRepository.findOneSmallerOrder(order);
+        if (findBallotTopic.size() == 0) {
+            return "";
```

Comment:
```suggestion
            return NO_RESULT;
```

상수로 선언하면 의미가 더 잘 표현될 것 같기도 ??? 이거는 미래의 우리가 다시 보면 헷갈릴 수도 있을 거같어